### PR TITLE
Fix fail to create three hops of ref pages

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/5570

Problem Summary: As described in https://github.com/pingcap/tiflash/issues/5570#issuecomment-1211734445

### What is changed and how it works?

In `VersionedPageEntries::incrRefCount` and `VersionedPageEntries::resolveToPageId` (with ignore_deletes=true), we ignore deletes before `ver` until we find an entry.
This is base on the assumption that we won't reuse the same page id before we cleanup all the ref to a page  in the upper layer.

tests:
* add segment logical split random test `SegmentOperationTest.SegmentLogicalSplit`
* add ref page random test `PageDirectoryTest.{NewRefAfterDelRandom,NewRefToExtAfterDelRandom}`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/PageDirectory.cpp [hunk 1]
```diff
@@ -988,9 +1040,38 @@ void PageDirectory::applyRefEditRecord(
...
-    // applying ref 3->2, existing ref 2->1, normal entry 1, then we should collapse
-    // the ref to be 3->1, increase the refcounting of normal entry 1
-    auto [resolve_success, resolved_id, resolved_ver] = [&mvcc_table_directory](PageIdV3Internal id_to_resolve, PageVersion ver_to_resolve)
+    // Assume the `mvcc_table_directory` is:
+    // {
+    //     "10": [
+    //         {
+    //             "type": "entry",
+    //             "create_ver": 1,
+    //             "being_ref_count": 3, // being ref by id 11, 12
```
Comment (lidezhu) [inline dbms/src/Storages/Page/V3/PageDirectory.cpp:1049]:
```suggestion
    //             "being_ref_count": 2, // being ref by id 11
```

**Review by lidezhu — APPROVED** [review]
LGTM with minor comment

**Review by flowbehappy — APPROVED** [review]
LGTM

**Pull request merged**